src/Workspaces/Core/MSBuild.BuildHost/IBuildHost.cs

```diff
@@ -20,12 +20,9 @@ internal interface IBuildHost
 
     Task<ImmutableArray<(string ProjectPath, string ProjectGuid)>> GetProjectsInSolutionAsync(string solutionFilePath, CancellationToken cancellationToken);
 
-    /// <summary>
-    /// Returns whether this project's is supported by this host.
-    /// </summary>
-    Task<bool> IsProjectFileSupportedAsync(string projectFilePath, CancellationToken cancellationToken);
+    Task<IRemoteProjectFile> LoadProjectFileAsync(string projectFilePath, string languageName, CancellationToken cancellationToken);
 
-    Task<IRemoteProjectFile> LoadProjectFileAsync(string projectFilePath, CancellationToken cancellationToken);
+    Task<string?> TryGetProjectOutputPathAsync(string projectFilePath, CancellationToken cancellationToken);
```

Comment:
what's an example of where this fails?

Reply:
The user could theoretically have a <ProjectFileReference> in their project to something very random and they have some custom MSBuild that would do something bad.